tests/tests_geomstats/test_pre_shape.py

```diff
@@ -391,21 +405,301 @@ def test_kendall_directional_curvature(self):
+        tg_vec_2 = space.to_tangent(vector[2], base_point)
+        hor_z = space.horizontal_projection(tg_vec_2, base_point)
+        tg_vec_3 = space.to_tangent(vector[3], base_point)
+        hor_h = space.horizontal_projection(tg_vec_3, base_point)
+
+        nabla_x_a_y_a_x_y, _, nabla_x_a_x_y, a_y_a_x_y, a_x_y = \
+            space.iterated_integrability_tensor_derivative_parallel(
+                hor_x, hor_y, base_point)
+
+        nabla_x_a_y_z, a_y_z = \
+            space.integrability_tensor_derivative_parallel(
+                hor_x, hor_y, hor_z, base_point)
+
+        nabla_x_a_x_y_2, a_x_y_2 = \
+            space.integrability_tensor_derivative_parallel(
+                hor_x, hor_x, hor_y, base_point)
+        self.assertAllClose(a_x_y, a_x_y_2)
+        self.assertAllClose(nabla_x_a_x_y, nabla_x_a_x_y_2)
+
+        nabla_x_a_y_x, a_y_x = \
+            space.integrability_tensor_derivative_parallel(
+                hor_x, hor_y, hor_x, base_point)
+
+        nabla_x_a_z_y, a_z_y = \
+            space.integrability_tensor_derivative_parallel(
+                hor_x, hor_z, hor_y, base_point)
+        self.assertAllClose(a_z_y, -a_y_z)
+        self.assertAllClose(a_x_y, -a_y_x)
+
+        # test scal(A_Y V, W)=0 with V = A_X Y and W = A_Y Z
+        result = scal(a_y_a_x_y, a_y_z)
+        self.assertAllClose(result, gs.zeros_like(result))
+
+        # Test scal{\nabla^S_X (A_Y V)}{W}  + scal{A_Y V}{nabla_X W} =0
+        # with V = A_X Y and W = A_Y Z
+        result = \
+            scal(nabla_x_a_y_a_x_y, a_y_z) \
+            + scal(a_y_a_x_y, nabla_x_a_y_z)
+        self.assertAllClose(result, gs.zeros_like(result))
+
+        # test scal(A_Y V, W)=0 with V = A_X Y and W = A_Z H
+        nabla_x_a_z_h, a_z_h = \
+            space.integrability_tensor_derivative_parallel(
+                hor_x, hor_z, hor_h, base_point)
+        result = scal(a_y_a_x_y, a_z_h)
+        self.assertAllClose(result, gs.zeros_like(result))
+
+        result = \
+            scal(nabla_x_a_y_a_x_y, a_z_h) \
+            + scal(a_y_a_x_y, nabla_x_a_z_h)
+        self.assertAllClose(result, gs.zeros_like(result))
+
+        # Test \scal{\nabla^S_X (A_Y Z)}{Z} + \scal{A_Y Z}{ A_X Z} =0
+        a_x_z = space.integrability_tensor(hor_x, hor_z, base_point)
+        result = scal(nabla_x_a_y_z, hor_z) + scal(a_y_z, a_x_z)
+        self.assertAllClose(result, gs.zeros_like(result))
+
+        # Test \nabla_X^S ( A_Y Z) + \nabla_X^S ( A_Z Y) = 0
+        result = nabla_x_a_y_z + nabla_x_a_z_y
+        self.assertAllClose(result, gs.zeros_like(result))
+
+        # Test \nabla_X^S ( A_X Y) + \nabla_X^S ( A_Y X) = 0
+        result = nabla_x_a_x_y + nabla_x_a_y_x
+        self.assertAllClose(result, gs.zeros_like(result))
+
+        # test the identity \scal{\nabla^S_X (A_Y A_X Y)}{Z}
+        #  + \scal{A_Y A_X Y}{ A_X Z} +  \scal{\nabla^S_X ( A_Y Z) }{A_X Y}
+        #  + \scal{A_Y Z}{\nabla^S_X A_X Y} = 0
```

Comment:
comments

Reply:
Thanks. Clarified, simplified and condensed the message to keep as few comments as possible.

Reply:
Thanks for the hints. I am currently breaking the tests in smaller units with docstring to explain what is done. 

However, I definitely cannot write in the docstring the formulas that are tested with notations A, B, C for the tangent vectors: I need to used capital letters H, X, Y, Z for horizontal vector fields and U,V W for vertical ones to make sense with respect to the original publications (and A is the integrability tensor itself in standard notations).  
In this test-file, this is probably nor very important, but in the docstrings of the main class, the same problem arise and create an issue for mapping the uninformative variable names often inherited from their parent class to more meaningful mathematical notations. I guess that this could be raised as a more general issue to solve later after this PR.

Reply:
Tests were broken in smaller units.